gdal/ogr/ogrsf_frmts/gml/ogrgmldatasource.cpp

```diff
@@ -872,14 +872,24 @@ bool OGRGMLDataSource::Open( GDALOpenInfo *poOpenInfo )
                 {
                     GMLRegistryNamespace &oNamespace =
                         oRegistry.aoNamespaces[iNS];
-                    const char *pszNSToFind =
-                        CPLSPrintf("xmlns:%s", oNamespace.osPrefix.c_str());
-                    const char *pszURIToFind =
-                        CPLSPrintf("\"%s\"", oNamespace.osURI.c_str());
+                    // When namespace is omitted or fit with case sensitive match for
+                    // name space prefix, then go next to find feature match.
+                    //
```

Comment:
Here I've change more than just adding a GFD logic, but refactoring it in small place.
An open()  for gml data is too huge so I think it is good chance to consider a design.
Actually open() does not only open also detecting a gml variant.